pkg/config/remote_config.go

```diff
@@ -0,0 +1,51 @@
+package config
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"cloud.google.com/go/firestore"
+)
+
+type RemoteConfig interface {
+	GetRemoteConfigTo(ctx context.Context, data interface{}) error
+
+	UpdateRemoteConfig(ctx context.Context, configPath string, value interface{}) error // configPath represent the fields that reference a value, accepts simpler form of field path as a string in which the individual fields are separated by '/' as the configPath
```

Comment:
Here, the configPath represents the path to target the specific field in the remote config.

Reply:
My main concern of this PR is whether the interface is generic which can provide extensibility of other implementations besides firestore.